packages/cli/src/util/client.ts

```diff
@@ -132,7 +153,97 @@ export default class Client extends EventEmitter implements Stdio {
     });
   }
 
-  private _fetch(_url: string, opts: FetchOptions = {}) {
+  /**
+   * When the auth config is of type `OAuthAuthConfig`,
+   * this method silently tries to refresh the access_token if it is expired.
+   *
+   * If the refresh_token is also expired, it will not attempt to refresh it.
+   * If there is any error during the refresh process, it will not throw an error.
+   */
+  private async ensureAuthorized(): Promise<void> {
+    if (!isOAuthAuth(this.authConfig)) return;
+
+    const { authConfig } = this;
+
+    // If we have a valid access token, do nothing
+    if (isValidAccessToken(authConfig)) {
+      output.debug('Valid access token, skipping token refresh.');
+      return;
+    }
+
+    // If we don't have a valid refresh token, do nothing
+    if (!isValidRefreshToken(authConfig)) {
+      output.debug('Invalid refresh token, skipping token refresh.');
+      return;
+    }
+
+    const tokenResponse = await refreshTokenRequest({
+      refresh_token: authConfig.refreshToken,
+    });
+
+    const [tokenError, token] = await processTokenResponse(tokenResponse);
+
+    // If we had an error, during the refresh process, silently continue
+    if (tokenError) {
+      output.debug('Error refreshing token, skipping token refresh.');
+      if (output.isDebugEnabled()) output.prettyError(tokenError);
+      return;
+    }
+
+    const [accessTokenError] = await verifyJWT(token.access_token);
+    if (accessTokenError) {
+      output.debug('Error verifying access token, skipping token refresh.');
+      if (output.isDebugEnabled()) output.prettyError(accessTokenError);
+      return;
+    }
```

Comment:
Is it possible to delete the `verifyJWT` code and let the client treat the access token and refresh token as opaque strings? By depending on the format here, it will be harder to change things in the future.